### PR TITLE
ARROW-2673: [Python] Add documentation + docstring for ARROW-2661

### PR DESCRIPTION
**Pull request by zeroshade**
Adding extra_conf arg documentation to the docstring

### CODE REVIEW

File: python/pyarrow/hdfs.py [hunk 1]
```diff
@@ -168,6 +168,8 @@ def connect(host="default", port=0, user=None, kerb_ticket=None,
     driver : {'libhdfs', 'libhdfs3'}, default 'libhdfs'
       Connect using libhdfs (JNI-based) or libhdfs3 (3rd-party C++
       library from Apache HAWQ (incubating) )
+    extra_conf : dict of extra Key/Value pairs for config; Will override
```
Comment (wesm) [inline python/pyarrow/hdfs.py]: This should be "dict, default None", and then the rest of the documentation go on the following line

**Review by wesm — COMMENTED** [review]
Thanks for the contribution. I think hdfs-site.xml may also be used with libhdfs without having an env variable set

File: python/pyarrow/hdfs.py (hunk 1 above)
Comment (zeroshade) [reply python/pyarrow/hdfs.py]: Fixed with the most recent commit :)

**Comment by zeroshade** [comment]
Can the build on this be restarted? I don't believe the failures here are related at all to my change.

**Comment by wesm** [comment]
Can you rebase? That will trigger a new build

**Comment by wesm** [comment]
Just rebased the branch

**Review by wesm — APPROVED** [review]
+1

**Pull request closed without merging**